engine.fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/RestFhirRetrieveProvider.java

```diff
@@ -26,15 +29,32 @@
 	private static final SearchStyleEnum DEFAULT_SEARCH_STYLE = SearchStyleEnum.GET;
 
 	protected IGenericClient fhirClient;
-	private SearchStyleEnum searchStyle;
+    private SearchStyleEnum searchStyle;
+
+    private CapabilityStatementIndex index;
 
 	public RestFhirRetrieveProvider(SearchParameterResolver searchParameterResolver, IGenericClient fhirClient) {
 		super(searchParameterResolver);
 		// TODO: Figure out how to validate that the searchParameterResolver and the
 		// client are on the same version of FHIR.
 		this.fhirClient = fhirClient;
-		this.searchStyle = DEFAULT_SEARCH_STYLE;
-	}
+        this.searchStyle = DEFAULT_SEARCH_STYLE;
+    }
+
+    @Override
+    protected synchronized CapabilityStatementIndex getIndex() {
+        if (this.getIndex() == null) {
```

Comment:
We are getting a StackOverflowError from this line. It looks like a bug to me. The code should be this.index == null not this.getIndex().